File: examples/sorting-algorithms/case-medium-random.ts
function lcg(seed: number): () => number {
  let state = seed;
  return () => {
    state = (state * 1664525 + 1013904223) & 0xffffffff;
    return (state >>> 0) / 0xffffffff;
  };
}

function randomArray(size: number, seed: number): number[] {
  const rng = lcg(seed);
  return Array.from({ length: size }, () => Math.floor(rng() * size * 10));
}

const data = randomArray(1000, 123);

export function createCase() {
  return {
    case: {
      caseId: "medium-random",
      caseClass: "medium",
      name: "Medium Random Array",
      version: "1.0.0",
      inputs: { data },
    },
    getInput: async () => ({ data }),
    getInputs: () => ({ data }),
  };
}

File: examples/sorting-algorithms/case_medium_random.py
"""Medium Random Array case — 1000 elements, LCG seed 123."""

import math


def _lcg(seed):
    state = seed

    def next_val():
        nonlocal state
        state = (state * 1664525 + 1013904223) & 0xFFFFFFFF
        return (state & 0xFFFFFFFF) / 0xFFFFFFFF

    return next_val


def _random_array(size, seed):
    rng = _lcg(seed)
    return [math.floor(rng() * size * 10) for _ in range(size)]


_data = _random_array(1000, 123)


def create_case():
    return {
        "case": {
            "caseId": "medium-random",
            "caseClass": "medium",
            "name": "Medium Random Array",
            "version": "1.0.0",
            "inputs": {"data": _data},
        },
        "getInput": lambda: {"data": _data},
        "getInputs": lambda: {"data": _data},
    }
